### DISCUSSION TITLE
Incorporate combat mode from /tg/station.

### DISCUSSION THREAD

**Discussion by PetMudstone** [unitystation/unitystation#7970] [Ideas]
### Description

This discussion is for current [/tg/'s combat mode ](https://github.com/tgstation/tgstation/pull/56601) and the possibility of importing it to Unitystation and the benefits of doing so.

The linked PR explains sufficiently enough I feel, but for sake of summary:
Combat mode replaces the four intent system. Combat mode is in practice a binary help/harm intent system. If it's off, you help. If it's on, you harm.

Disarm and grab intents are gone, but their function still remains. Everything grab related is now done via control + click. Meanwhile, disarms are done via right clicking on people with an empty hand regardless of if combat mode is on or off.

Speaking of right click, the context menu is no longer on right click and is now (always) on shift + right click. This frees up right click to be used as an alternative action without arbitrarily removing the context menu, such as with how disarms work as mentioned above. For example, you disassemble racks and tables with tools using right click instead of using left click on harm intent. This enables you to put tools on tables at any time. This also allows us to get rid of various awkward item toggles. such as syringes needing to be toggled between inject and draw modes in old /tg/. Now, you inject with syringes using left click and draw using right click.

Combat mode should be easier to understand by new players over the odd 4 intent system. Right clicks give us increased QoL in regards to implementing alternate interactions. Having spoken to various /tg/ and downstream players, they don't mind the changes from combat mode at all and often appreciate them. I myself wasn't hindered while testing out combat mode and appreciate the QoL changes that came with right click interactions.

**Comment by wraith-54321** [comment]
I personally like combat mode but I could see why people would not want to switch

**Reply by PetMudstone** [reply]
Yes, I would like to hear arguments against combat mode.

**Comment by Bod9001** [comment]
just turn the harm into combat mode, and then keep the rest of the intense

**Reply by PetMudstone** [reply]
As in keeping the 4 intent system as we have now?

**Reply by Bod9001** [reply]
just the buttons for them. since the code for help, drag,  disarm stays exactly the same

**Reply by PetMudstone** [reply]
Keeping disarm and grab as distinct intents defeats the point of combat mode in streamlining the system, I feel. I suppose we could keep 4 intents and still move the context menu to shift + right click, but I don't really get why we wouldn't want to go the whole way.

**Comment by corp-0** [comment]
I also think combat mode is the way to go and don't see any counter argument other than being way too used to 4 intents.

**Reply by PetMudstone** [reply]
Yeah, the main arguments I've seen have largely boiled down to people being very familiar with the 4 intent system and also right click being the context menu in not just SS13, but in basically everywhere else.

**Comment by Pockets-byte** [comment]
I really don't see the point in combat mode, it makes the game less flexible, and it's much less clear on how to use than intents
I figured out intents instantly, but I still don't know how combat mode works, aside from needing a wiki to use, having the entire control scheme randomly change at the drop of a hat isn't good either, as that's directly opposite of what's needed for muscle memory and learning the controls
also we were trying to recreate classic tg right? so why combat mode

**Reply by PetMudstone** [reply]
Combat mode did means any interactions in /tg/ that relied on disarm and grab mode now use right click and other methods. For example, deliberately botching surgeries required disarm intent. Now you just right click. I do not see this as a bad thing at all. Regardless, right clicking as an alternative action should give us a lot more options, as stated in my main post.

I don't think it's much less clear to use than intents. I've looked at the /tg/ wiki pages for both [combat mode](https://tgstation13.org/wiki/Combat_Mode) and the [/tg/'s combat mode](https://tgstation13.org/wiki/Intent) and they are almost identical. I feel like it's likely easier to explain a two intent system with disarm and grab on keybinds that are available at all times than a four intent system where two of the intents are highly situational and unnecessary in except in very particular circumstances.

Also, combat mode doesn't change the entire control scheme at random, what are you talking about? Combat mode is basically just harm intent and works as harm intent did as before. Grabbing*, disarming, and the context menu still work the same regardless of if you're in combat mode or not.

*Control clicking on an empty floor while combat mode is not enabled in /tg/ does release whoever you are grabbing, but that's it.

**Comment by CdrCross** [comment]
My suggestion would be that instead of disarm being omnipresent on both help and harm intent, it should be the right-click option when clicking on characters and mobs in combat mode. Think about it: Most of your radial interactions on characters belong under help, so disabling the right click radial in combat mode doesn't actually take anything away, and in turn enables much quicker hopping back and forth between lethal attacks and disarming attempts.

**Reply by PetMudstone** [reply]
If you mean making right click on Help bring up the radial menu while on Harm it does disarms, I must greatly disagree. As stated previously Shift + Right click always brings up the context menu in /tg/ now, and for us it would always bring up the radial menu no matter what. 

This keeps things consistent and less confusing, as as there won't be any strange changes in interaction whole switching between modes. It must also be remembered that taking the context/radial menu off right click frees it up to be used as an alternate interaction on/for many different items and objects, which I honestly believe is the most important part of the combat mode rework.

**Reply by CdrCross** [reply]
At least until we have proper tutorial tools in place, I am heavily inclined towards leaning on the UI conventions of operating systems and other widespread software to help make learning our interface frictionless. Shift+RMB fucks with that, and so I am very hesitant about that until we have a means to properly teach people how our controls differ from the "default".

**Reply by PetMudstone** [reply]
I don't think it's too hard to learn the difference. I got used to it real quick. In fact, even in servers without combat mode I still instinctively hold shift while using right click even though it doesn't do anything while bringing up the context menu.

Also devs put context menu on Help right click and disarm on Harm right click before and people really didn't like it.
